test/test_automl/test_post_fit.py

```diff
@@ -66,3 +69,32 @@ def test_no_ensemble(automl: AutoML) -> None:
     assert automl.ensemble_ is None
     assert len(automl.models_) == 0
     assert len(automl.cv_models_) == 0
+
+
+@parametrize_with_cases("automl", cases, has_tag=["multiobjective"])
+def test__load_pareto_front(automl: AutoML) -> None:
+    """
+    Parameters
+    ----------
+    automl : AutoML
+        An AutoML object fitted with multiple objective metrics
+
+    Expects
+    -------
+    * Auto-sklearn can predict and has a model
+    * _load_pareto_front returns one scikit-learn ensemble
+    """
+    # Check that the predict function works
+    X = np.array([[1.0, 1.0, 1.0, 1.0]])
+    print(automl.predict(X))
+    assert automl.predict_proba(X).shape == (1, 3)
+    assert automl.predict(X).shape == (1,)
+
+    pareto_front = automl._load_pareto_front()
+    assert len(pareto_front) == 1
+    for ensemble in pareto_front:
+        assert isinstance(ensemble, (VotingClassifier, VotingRegressor))
+        y_pred = ensemble.predict_proba(X)
+        assert y_pred.shape == (1, 3)
+        y_pred = ensemble.predict(X)
+        assert y_pred in ["setosa", "versicolor", "virginica"]
```

Comment:
It's not very clear why this should be only one scikit learn ensemble expected but I assume it's because of the default parameter for ensemble selection.

It also seems this test is very specific to this single case (multiobjective iris classifier).

I had the same problem when considering cases and my solution was just to have general tests. We can just push this through for now, knowing it will break if we add any other cases with the `"multiobjective"` tag.

The longer term solution to this, I have a few ideas:
* We just use `make_automl`, `make_dataset` and construct the specific automl instance for this test where the specifics that are being tested are directly evident in the test.
* We encode these extra specifics somehow:
    * The case just returns extra info
    ```python
    def case_classifier_fitted_holdout_multiobjective(...):
        ...
        return (model, extra_info)
    ```
    * The extra specifics are directly saved and access on the model object. This does add a lot more introspection capabilites to the model which may be helpful for future additions
    
Happy to hear any other ideas on this though, I admit the caching solution as is, is not perfect for this reason, but it does allow the tests to be a lot more modular.

Reply:
> Would we for the 2nd idea check whether the AutoML was built on iris and then use it? Besides that, could we maybe add a filter on which dataset(s) were used to build the AutoML system?

Yup it's definitely possible, the easiest way is to just do so in the test itself, i.e. `if extra_info["dataset"] != "iris": pass` but I'm not the biggest fan of the solution.

The overarching problem is that you can't use `@parametrize` and `@tags` together, i.e. you can't associate a parameter with a tag.

I guess my prefered solution is to include more general things in the `extra_info` or encode it on the model, meaning the tests don't have to do any filtering.

```python
extra_info = {
    "X_shape": X.shape,
    "y_shape": y.shape,
    "labels": ...
}
return (automl, extra_info)
```

It's not the cleanest but at least it means this test could theoretically work for any other "multiobjective" tagged case, as long as it provides the necessary extra_info.